pkg/cmd/preview/preview.go

```diff
@@ -865,10 +865,11 @@ func (o *PreviewOptions) DefaultValues(ns string, warnMissingName bool) error {
 			}
 		}
 	}
-	o.Name = naming.ToValidName(o.Name)
 	if o.Name == "" {
 		return fmt.Errorf("No name could be defaulted for the Preview Environment. Please supply one!")
 	}
+	o.Name = naming.ToValidName(o.Name)
```

Comment:
Is moving this code not just patching up an underlying problem in `naming.ToValidName`? The semantics used to be that `ToValidName` of the empty string is the empty string, but now it is `x`. Whether the original semantics is correct might be arguable, but returning 'x' seems even worse. 

So imo to fix the issue we should change `naming.ToValidName` and handle empty string explicitly. While on it we could add a test for this case and update the docs.